velox/expression/EvalCtx.h

```diff
@@ -319,6 +319,43 @@ struct ScopedContextSaver {
   EvalCtx::ErrorVectorPtr errors;
 };
 
+/// Produces a SelectivityVector with a single row selected using a pool of
+/// SelectivityVectors managed by the EvalCtx::execCtx().
+class LocalSingleRow {
+ public:
+  LocalSingleRow(EvalCtx& context, vector_size_t row)
+      : context_(*context.execCtx()),
+        vector_(context_.getSelectivityVector(row + 1)) {
+    vector_->clearAll();
+    vector_->setValid(row, true);
+    vector_->updateBounds();
+  }
+
+  ~LocalSingleRow() {
+    context_.releaseSelectivityVector(std::move(vector_));
+  }
+
+  SelectivityVector& operator*() {
+    return *vector_;
+  }
+
+  const SelectivityVector& operator*() const {
+    return *vector_;
+  }
+
+  SelectivityVector* operator->() {
+    return vector_.get();
+  }
+
+  const SelectivityVector* operator->() const {
+    return vector_.get();
+  }
+
+ private:
+  core::ExecCtx& context_;
+  std::unique_ptr<SelectivityVector> vector_;
```

Comment:
The memory management for this was a bit non-intuitive to me since I expected to see a shared_ptr here given this was from a pool of reusable SelectivityVector from the context. But it makes sense for this to be a unique_ptr to be released back also assuming the context outlasts this class. Please can you add a comment about this to make it more clear.